src/common/scratch_buffer.h

```diff
@@ -87,6 +90,18 @@ class ScratchBuffer {
         return buffer_capacity;
     }
 
+    void swap(ScratchBuffer& other) {
+        std::swap(last_requested_size, other.last_requested_size);
+        std::swap(buffer_capacity, other.buffer_capacity);
+        std::swap(buffer, other.buffer);
+    }
+
+    void swap(ScratchBuffer&& other) {
```

Comment:
```suggestion
    void swap(ScratchBuffer&& other) noexcept {
```

Shouldn't this overload reset the other instance to non-valid/default values rather than potentially giving it a valid buffer? If no, then you should only need the regular by-reference overload above